### ISSUE TITLE
Crashes on yt livestreams

### ISSUE THREAD

**Issue by mkg20001** [tizonia/tizonia-openmax-il#403]
The first song in queue is likely a livestream. After running the command below tizonia crashed.

```
maciej@maciej-pc:~$ tizonia --youtube-audio-mix-search "Monstercat Music"
tizonia 0.10.0. Copyright (C) 2017 Juan A. Rubio
This software is part of the Tizonia project <http://tizonia.org>

[YouTube] [Stream] '24/7 Monstercat Radio - Gaming | Study | Relax - Electronic Dance Live Stream'.
[YouTube] [Stream] 'Monstercat - Best of 2016 (Album Mix) [2.5 Hours of Electronic Music]'.
[YouTube] [Stream] 'Best Gaming Music Mix 2017 | Dubstep, Trap, Drumstep, Electro'.
[YouTube] [Stream] 'Tokyo Machine - ROCK IT [Monstercat Release]'.
[YouTube] [Stream] 'Marshmello - Alone [Monstercat Official Music Video]'.
[YouTube] [Stream] 'Ultimate Gaming Music Mix 2016 Electro House Dubstep Drops Drumstep #18'.
[YouTube] [Stream] 'Best Music Mix 2016 |  1H Gaming Music  | Dubstep, Electro House, EDM, Trap'.
[YouTube] [Stream] 'Gaming Music Mix 2017  Best Dubstep  Electro House  Trap & Bass'.
[YouTube] [Stream] 'Monstercat Uncaged - Vol. 2 (Album Mix)'.
[YouTube] [Stream] 'Best Gaming Music Mix   (1 HOUR) DUBSTEP, TRAP, EDM, BASS - Copyright / Royalty Free Music NCS'.
[YouTube] [Stream] 'Best Music Mix 2016 |  1H Gaming Music  | Dubstep, Electro House, EDM, Trap'.
[YouTube] [Stream] '[Drumstep] - Tristam & Braken - Flight [Monstercat Release]'.
[YouTube] [Stream] 'Dubstep Gaming Music  Best Dubstep, Drum n Bass, Drumstep  It's Gaming Time'.
[YouTube] [Stream] 'Dubstep Gaming Music - Best of EDM - Live stream 24/7 | Electro/House/Dubstep Drops/Drumstep'.
[YouTube] [Stream] 'Slushii - LUV U NEED U [Monstercat Release]'.
[YouTube] [Stream] 'Trap Music Mix 2017  Suicide Squad Trap  Trap & Bass | Best EDM'.
[YouTube] [Stream] 'Best of Electronic Music - Vol.1 (1 Hour Mix) [Monstercat Release]'.
[YouTube] [Stream] 'Rocket League x Monstercat Vol. 1 (Album Mix)'.
[YouTube] [Stream] 'Best Music Mix 2017 | Best of EDM | NoCopyrightSounds x Gaming Music'.
[YouTube] [Stream] 'Electro Music - Electronic Dance Music 2017  Charts 2017 top 40 summer mix 2017 edc'.
[YouTube] [Stream] 'Tristam - Bone Dry [Monstercat Release]'.
[YouTube] [Stream] 'Best Electronic Dance Music 2015 -  electronic dance music'.
[YouTube] [Stream] 'Monstercat Uncaged - Vol. 1 (Album Mix)'.
[YouTube] [Stream] 'New Electro & House 2013 Best Of EDM Mix'.
[YouTube] [Stream] '1 HOUR BEST GAMING MIX {ELECTRO, HARD DANCE, DUBSTEP, DRUMSTEP}'.
[YouTube] [Stream] 'Rogue - Fury [Monstercat Release]'.
[YouTube] [Streams in queue] '26'.
tizonia: webmdmuxfltprc.c:669: demux_stream: Assertion `ap_prc->p_ne_' failed.
Abgebrochen (Speicherabzug geschrieben)
```
`Abgebrochen (Speicherabzug geschrieben)` = Coredumped

**Comment by juanrubio** [comment]
Thanks! 

Very interesting. There are two issues here:
# libnestegg (on which tizonia's webm demuxer is based) is unable to properly parse that stream.
# tizonia's webm demuxer incorrectly ignores the  issue leading to a crash

For the second issue, I'll produce a fix that simply ignore this stream and skip to the next one. For the first issue, I will need to dig more. It could be a libnestegg issue.

**Comment by juanrubio** [comment]
> libnestegg (on which tizonia's webm demuxer is based) is unable to properly parse that stream.
I realized that the stream is parsed correctly unless sufficient data is provided (not a problem in nestegg)

> tizonia's webm demuxer incorrectly ignores the issue leading to a crash
This is confirmed and will be fixed in #407

Finally, there is a bit of work to do in the Tizonia 'player' to handle the situation, namely, the webm demuxer fails to demux the stream (for whatever reason) and the youtube audio player state machine needs to gracefully react by skipping the stream and processing the next in the queue. This will be fixed here.

**Comment by juanrubio** [comment]
All right, I'm closing this issue now. With the last few commits in master the crash is now avoided because the live stream in the playlist queue (the first item in the example reported) is detected and simply discarded. The remaining streams in the queue play back correctly.

So the only issue remaining is the live stream itself that will be played back when #391 and #392 are fixed.

Thanks for reporting. This has been a really interesting issue to fix!.
